### PR TITLE
feat(translation): Add video translation.

### PR DESCRIPTION
**Pull request by IRONICBo**
Support video translation.

1. Use QiNiu ASR(automatic speech recognition) service to convert audio to text.
2. Use TTS service to generation audio from text.

### CODE REVIEW

File: translation/video_test.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	formUploader := storage.NewFormUploaderEx(&cfg, &storage.Client{Client: &client})
+	ret := storage.PutRet{}
+	err := formUploader.PutFile(context.Background(), &ret, upToken, key, localFile, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Log(ret)
+}
+
+func TestASR(t *testing.T) {
```
Comment (IRONICBo) [inline translation/video_test.go:65]: Request api error.

File: translation/video_test.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	}
+
+	mac := auth.New(mockAccessKey, mockSecretKey)
+	bearToken, err := mac.SignRequestV2(req)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", bearToken))
```
Comment (CarlJi) [inline translation/video_test.go:139]: `Bearer` 替换成 `Qiniu` 试试

File: translation/video_test.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+		t.Fatal(err)
+	}
+
+	mac := auth.New(mockAccessKey, mockSecretKey)
+	bearToken, err := mac.SignRequestV2(req)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	req.Header.Set("Content-Type", "application/json")
```
Comment (CarlJi) [inline translation/video_test.go:138]: `Content-Type` 也要签进去

File: translation/markdown.go [hunk 4]
```diff
@@ -63,9 +65,11 @@ type translateResponse struct {
 }
 
 // New create a new TranslateConfig
-func New(apiKey string) *Engine {
+func New(translationAPIKey string, qiniuAccessKey string, qiniuSecretKey string) *Engine {
```
[the remark this answers is not in the document]
Comment (IRONICBo) [reply translation/markdown.go:70]: `*Config` is the same as `Engine` struct ?

**Comment by IRONICBo** [comment]
Test ASR Result:

```bash
resp: &translation.ASRTaskData{Rtn:0, Message:"SAAS_SUCC", RequestId:"28a5227559b64aaebbf256762d3e4b83", TaskId:"eca1b15443c54b0b8439d6ac0f44ce41", Data:translation.Data{StatusCode:3, StatusText:"TASK_SUCC", SpeechResult:translation.SpeechResult{ResultText:"上大学以后，我当时没有条件，只能宿舍的舍友吧就五个人买合买了一台电脑，五分，只有1/5的时间我能碰电脑，所以我基本上编程的时候就会在纸上先把所有你的思路全部写好，大部人会喜欢在电脑前把代码敲出来，然后一步步单步跟踪调试，而我是比较少这么做的，所以我是觉得应该把自己逼到一种资源相对匮乏的情况下，你去去让自己突破，其实是更更好的一件事情。", Duration:35386, Detail:[]translation.Detail{translation.Detail{Sentences:"上大学以后，", StartTime:"200", EndTime:"1000", SpeakerId:"", WordsPiece:[]translation.WordsPiece(nil)}, translation.Detail{Sentences:"我当时没有条件，", StartTime:"1000", EndTime:"2720", SpeakerId:"", WordsPiece:[]translation.WordsPiece(nil)}, translation.Detail{Sentences:"只能宿舍的舍友吧就五个人买合买了一台电脑，", StartTime:"2720", EndTime:"6020", SpeakerId:"", WordsPiece:[]translation.WordsPiece(nil)}, translation.Detail{Sentences:"五分，", StartTime:"6360", EndTime:"6680", SpeakerId:"", WordsPiece:[]translation.WordsPiece(nil)}, translation.Detail{Sentences:"只有1/5的时间我能碰电脑，", StartTime:"6680", EndTime:"8880", SpeakerId:"", WordsPiece:[]translation.WordsPiece(nil)}, translation.Detail{Sentences:"所以我基本上编程的时候就会在纸上先把所有你的思路全部写好，", StartTime:"8880", EndTime:"14680", SpeakerId:"", WordsPiece:[]translation.WordsPiece(nil)}, translation.Detail{Sentences:"大部人会喜欢在电脑前把代码敲出来，", StartTime:"14680", EndTime:"17520", SpeakerId:"", WordsPiece:[]translation.WordsPiece(nil)}, translation.Detail{Sentences:"然后一步步单步跟踪调试，", StartTime:"17520", EndTime:"19380", SpeakerId:"", WordsPiece:[]translation.WordsPiece(nil)}, translation.Detail{Sentences:"而我是比较少这么做的，", StartTime:"20240", EndTime:"21800", SpeakerId:"", WordsPiece:[]translation.WordsPiece(nil)}, translation.Detail{Sentences:"所以我是觉得应该把自己逼到一种资源相对匮乏的情况下，", StartTime:"21800", EndTime:"26880", SpeakerId:"", WordsPiece:[]translation.WordsPiece(nil)}, translation.Detail{Sentences:"你去去让自己突破，", StartTime:"26880", EndTime:"29120", SpeakerId:"", WordsPiece:[]translation.WordsPiece(nil)}, translation.Detail{Sentences:"其实是更更好的一件事情。", StartTime:"29120", EndTime:"31660", SpeakerId:"", WordsPiece:[]translation.WordsPiece(nil)}}}}}PASS
ok      github.com/goplus/community/translation 0.893s
```

**Comment by IRONICBo** [comment]
### Raw WebVTT
```bash
WEBVTT FILE

1
00:00:00.200 --> 00:00:01.000
上大学以后，

2
00:00:01.000 --> 00:00:02.720
我当时没有条件，

3
00:00:02.720 --> 00:00:06.020
只能宿舍的舍友吧就五个人买合买了一台电脑，

4
00:00:06.360 --> 00:00:06.680
五分，

5
00:00:06.680 --> 00:00:08.880
只有1/5的时间我能碰电脑，

6
00:00:08.880 --> 00:00:14.680
所以我基本上编程的时候就会在纸上先把所有你的思路全部写好，

7
00:00:14.680 --> 00:00:17.520
大部人会喜欢在电脑前把代码敲出来，

8
00:00:17.520 --> 00:00:19.380
然后一步步单步跟踪调试，

9
00:00:20.240 --> 00:00:21.800
而我是比较少这么做的，

10
00:00:21.800 --> 00:00:26.880
所以我是觉得应该把自己逼到一种资源相对匮乏的情况下，

11
00:00:26.880 --> 00:00:29.120
你去去让自己突破，

12
00:00:29.120 --> 00:00:31.660
其实是更更好的一件事情。
```

### Translated WebVTT
```bash
WEBVTT FILE

1
00:00:00.200 --> 00:00:01.000
After going to college,

2
00:00:01.000 --> 00:00:02.720
I didn't have the conditions,

3
00:00:02.720 --> 00:00:06.020
Only the roommates in the dormitory bought a computer for five people.

4
00:00:06.360 --> 00:00:06.680
Five points,

5
00:00:06.680 --> 00:00:08.880
I can only touch the computer 1/5 of the time,

6
00:00:08.880 --> 00:00:14.680
So when I basically program, I will write all your ideas on paper first.

7
00:00:14.680 --> 00:00:17.520
Most people like to type out the code in front of the computer.

8
00:00:17.520 --> 00:00:19.380
Then trace the debugging step by step,

9
00:00:20.240 --> 00:00:21.800
And I do this less often.

10
00:00:21.800 --> 00:00:26.880
Therefore, I feel that I should push myself into a situation where resources are relatively scarce.

11
00:00:26.880 --> 00:00:29.120
You go and let yourself break through,

12
00:00:29.120 --> 00:00:31.660
It's actually a better thing.
```

**Comment by IRONICBo** [comment]
Video player demo with caption.
```html
<!DOCTYPE html>
<html>
<head>
    <link rel="stylesheet" href="https://cdn.plyr.io/3.7.8/demo.css">
    <script src="https://cdn.plyr.io/3.6.8/plyr.js"></script>
</head>
<body>
    <video id="player" controls crossorigin playsinline data-plyr-config='{ "title": "Example" }'>
        <source src="http://s7cjwtxc2.hn-bkt.clouddn.com/test.mp4" type="video/mp4" />
        <track kind="captions" label="English" srclang="en" src="http://s7cjwtxc2.hn-bkt.clouddn.com/test_en.vtt" default>
        <track kind="captions" label="Origin" srclang="origin" src="http://s7cjwtxc2.hn-bkt.clouddn.com/test.vtt">
    </video>

    <script>
      document.addEventListener('DOMContentLoaded', function() {
        const player = new Plyr('#player');
    });
    </script>
</body>
</html>
```

**Comment by IRONICBo** [comment]
@CarlJi  Please review :)

File: internal/core/community.go [hunk 5]
```diff
@@ -83,9 +83,8 @@ type CasdoorConfig struct {
 }
 
 type Account struct {
-	casdoorConfig *CasdoorConfig
-
-	xLog *xlog.Logger
+	// casdoorConfig *CasdoorConfig
```
Comment (CarlJi) [inline internal/core/community.go]: recommend to remove useless code

File: internal/core/media.go [hunk 6]
```diff
@@ -4,14 +4,15 @@ import (
...
-	"github.com/qiniu/x/xlog"
 	"io"
 	"os"
 	"strconv"
 	"strings"
 	"time"
 
+	"github.com/goplus/yap"
+	"github.com/qiniu/x/xlog"
+
```
Comment (CarlJi) [inline internal/core/media.go]: remove this space line

File: translation/video.go [hunk 7]
```diff
@@ -161,3 +204,204 @@ func (e *Engine) Text2Audio(content string) (*TTSResponse, error) {
...
+		// Skip empty line
+		if line == "" {
+			continue
+		}
+
+		// Read batch content
+		if fs := strings.Fields(line); len(fs) > 0 {
+			// Read index
+			index, err := strconv.Atoi(fs[0])
+			if err != nil {
```
Comment (CarlJi) [inline translation/video.go:332]: recommend to add log here in order to record exception

Comment (IRONICBo) [reply translation/video.go:332]: I try to throw an error in this funciton.

**Review by CarlJi — APPROVED** [review]
LGTM

**Pull request merged**